File: 1493. Longest Subarray of 1's After Deleting One Element/1493. Longest Subarray of 1's After Deleting One Element.ts
function longestSubarray(nums: number[]): number {
	const n = nums.length;
	let left = 0,
		right = 0;
	let zeroCount = 0;
	let maxLen = 0;

	while (right < n) {
		if (nums[right] === 0) {
			zeroCount++;
		}

		while (zeroCount > 1) {
			if (nums[left] === 0) {
				zeroCount--;
			}
			left++;
		}

		maxLen = Math.max(maxLen, right - left);
		right++;
	}

	return maxLen;
}

File: 1493. Longest Subarray of 1's After Deleting One Element/1493. Longest Subarray of 1's After Deleting One Element.py
class Solution:
    def longestSubarray(self, nums: List[int]) -> int:
        n = len(nums)
        left, right = 0, 0
        zero_count = 0
        max_len = 0

        while right < n:
            if nums[right] == 0:
                zero_count += 1

            while zero_count > 1:
                if nums[left] == 0:
                    zero_count -= 1
                left += 1

            max_len = max(max_len, right - left)
            right += 1

        return max_len